x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/index.test.tsx

```diff
@@ -1391,4 +1391,202 @@ describe('when on the endpoint list page', () => {
       expect(startButton).toBeNull();
     });
   });
+  describe('endpoint list take action with RBAC controls', () => {
+    let renderResult: ReturnType<AppContextTestRender['render']>;
+
+    const renderAndClickActionsButton = async (tableRow: number = 0) => {
+      reactTestingLibrary.act(() => {
+        history.push(`${MANAGEMENT_PATH}/endpoints`);
+      });
+
+      renderResult = render();
+      await middlewareSpy.waitForAction('serverReturnedEndpointList');
+      await middlewareSpy.waitForAction('serverReturnedEndpointAgentPolicies');
+
+      const endpointActionsButton: HTMLElement = (
+        await renderResult.findAllByTestId('endpointTableRowActions')
+      )[tableRow];
+
+      reactTestingLibrary.act(() => {
+        reactTestingLibrary.fireEvent.click(endpointActionsButton);
+      });
+    };
+
+    beforeEach(async () => {
+      const { data: hosts } = mockEndpointResultList({ total: 2 });
+      // second host is isolated, for unisolate testing
+      const hostInfo: HostInfo[] = [
```

Comment:
Can we get this from the generator instead?

Reply:
So we do get most of the base stuff from the generator in `mockEndpointResultList`, but I just needed to ensure some of the capabilities / isolation states are fixed to test certain things